asmcomp/amd64/emit.mlp

```diff
@@ -649,6 +649,27 @@ let emit_instr fallthrough i =
         def_label label;
         I.lea (mem64 NONE 8 R15) (res i 0)
       end
+  | Lop(Ipoll { return_label }) ->
+      I.cmp (domain_field Domainstate.Domain_young_limit) r15;
+      let gc_call_label = new_label () in
+      let label_after_gc = match return_label with
+                  | None -> new_label()
+                  | Some(lbl) -> lbl in
+      let lbl_frame =
+        record_frame_label i.live (Dbg_alloc [])
+      in
+      begin match return_label with
+      | None -> I.jbe (label gc_call_label)
+      | Some return_label -> I.jg (label return_label)
+      end;
```

Comment:
The opposite of `jbe` is `ja` (unsigned comparison), and the opposite of `jg` is `jle` (signed comparison).  I think you want unsigned comparisons, like for `Ialloc`.